overlord/state/taskrunner.go

```diff
@@ -336,20 +336,23 @@ func (r *TaskRunner) Ensure() {
 			}
 			continue
 		}
+
+		if mustWait(t) {
+			// Dependencies still unhandled.
+			continue
+		}
+
 		if status == UndoStatus && handlers.undo == nil {
 			// Cannot undo. Revert to done status.
 			t.SetStatus(DoneStatus)
+			// We may have to wait for dependencies, as further
```

Comment:
Please move this to the first line inside this block, and let's clarify the comment a bit further:

// Although this has no dependencies itself, it must have waited
// above too since follow up tasks may have handlers again.